tests/integration/targets/android_sdk/tasks/setup.yml

```diff
@@ -0,0 +1,86 @@
+---
+####################################################################
+# WARNING: These are designed specifically for Ansible tests       #
+# and should not be used as examples of how to write Ansible roles #
+####################################################################
+
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Include OS-specific variables
+  include_vars: '{{ ansible_os_family }}.yml'
+
+- name: Install dependencies
+  become: true
+  package:
+    name:
+      - "{{ openjdk_pkg }}"
+      - unzip
+    state: present
+  when: ansible_os_family != 'Darwin'
+
+- name: Install dependencies (OSX)
+  block:
```

Comment:
In the future it might be interesting to refactor this out of this file and run code selectively per platform - similar to what you did with the variables above. The tricky part is that it will require a fallback ("default"?!) value for the selecting variable. This can be done later.